### ISSUE TITLE
Ensure all tests run and pass after refactor

### ISSUE THREAD

**Issue by vuldin** [redpanda-data/helm-charts#84]
When TLS is disabled, all associated non-TLS tests run and pass successfully. When TLS is enabled, the kafka test passes but there is an issue with the pandaproxy/REST TLS test. The associated section of the ConfigMap:
```yaml
    pandaproxy:
      pandaproxy_api:
      - address: 0.0.0.0
        name: internal
        port: 8082
      pandaproxy_api_tls:
      - name: internal
        enabled: true
        require_client_auth: true
        cert_file: /etc/tls/certs/cert1/tls.crt
        key_file: /etc/tls/certs/cert1/tls.key
        truststore_file: /etc/tls/certs/cert1/ca.crt
    schemaregistry:
      schemaregistry_api:
      - address: 0.0.0.0
        name: schema-registry
        port: 8081
      schemaregistry_api_tls:
      - name: schema-registry
        enabled: true
        require_client_auth: false
        cert_file: /etc/tls/certs/cert1/tls.crt
        key_file: /etc/tls/certs/cert1/tls.key
```
Running the same command as in the test container:
```sh
> k exec -it -n helm-test redpanda-0 -- curl -m 3 --ssl-reqd --cacert /etc/tls/certs/cert1/ca.crt https://redpanda-0.redpanda.helm-test.svc.cluster.local.:8082/brokers
Defaulted container "redpanda" out of: redpanda, redpanda-configurator (init)
curl: (56) OpenSSL SSL_read: error:1409445C:SSL routines:ssl3_read_bytes:tlsv13 alert certificate required, errno 0
command terminated with exit code 56
```

**Comment by vuldin** [comment]
More details on the current issue: TLS connection works on the kafka endpoints, but not on the rest/pandaproxy or schemaregistry endpoints:

Working test of kafka endpoint:
```sh
> k exec -it -n helm-test redpanda-0 -- openssl s_client -connect redpanda-0.redpanda.helm-test.svc.cluster.local.:9092 -CAfile /etc/tls/certs/cert1/ca.crt -tlsextdebug   
Defaulted container "redpanda" out of: redpanda, redpanda-configurator (init)
CONNECTED(00000003)                                                                   
TLS server extension "key share" (id=51), len=36                
0000 - 00 1d 00 20 dd f4 94 2e-98 78 40 9c c3 a8 f0 f7   ... .....x@.....
0010 - 01 a8 f1 19 0b 0a 95 d5-c7 76 3d 3b 70 97 ce 6d   .........v=;p..m
0020 - 70 c1 fa 05                                       p...   
TLS server extension "supported versions" (id=43), len=2        
0000 - 03 04                                             ..     
Can't use SSL_get_servername                                                          
depth=1 CN = redpanda-cert1-ca
verify return:1          
depth=0 CN = redpanda-cert1-cert
verify return:1
---                          
Certificate chain
 0 s:CN = redpanda-cert1-cert
   i:CN = redpanda-cert1-ca        
---                        
Server certificate        
-----BEGIN CERTIFICATE-----      
MIICSjCCAfCgAwIBAgIRAPMIk5z1Tnl4k9tnupg57Y4wCgYIKoZIzj0EAwIwHDEa
MBgGA1UEAxMRcmVkcGFuZGEtY2VydDEtY2EwHhcNMjIwNjE1MDUzMjMwWhcNMjcw
NjE0MDUzMjMwWjAeMRwwGgYDVQQDExNyZWRwYW5kYS1jZXJ0MS1jZXJ0MFkwEwYH
KoZIzj0CAQYIKoZIzj0DAQcDQgAEdPw0lwHLXyONXckzgKlZ8+3QcptiTA5Tc+X9
4w4JlF7jC0bkHgZfzZIvWEcHYBjJ84K/OhCKLPDBbt481u7EIqOCAQ8wggELMA4G
A1UdDwEB/wQEAwIFoDAMBgNVHRMBAf8EAjAAMB8GA1UdIwQYMBaAFODKw06CZ+dV
PQ6XOMOuh+zkTvchMIHJBgNVHREEgcEwgb6CNXJlZHBhbmRhLWNsdXN0ZXIucmVk
cGFuZGEuaGVsbS10ZXN0LnN2Yy5jbHVzdGVyLmxvY2FsgjcqLnJlZHBhbmRhLWNs
dXN0ZXIucmVkcGFuZGEuaGVsbS10ZXN0LnN2Yy5jbHVzdGVyLmxvY2FsgiRyZWRw
YW5kYS5oZWxtLXRlc3Quc3ZjLmNsdXN0ZXIubG9jYWyCJioucmVkcGFuZGEuaGVs
bS10ZXN0LnN2Yy5jbHVzdGVyLmxvY2FsMAoGCCqGSM49BAMCA0gAMEUCIQD6AHow
BTxPuHrvyslr/TR3H60Id3w5+s5TbSbNovR/mAIgOGI43OfSQ81GRI33lozVNLfX
UuJTUaPFX0UNTeYDleA=
-----END CERTIFICATE-----    
subject=CN = redpanda-cert1-cert

issuer=CN = redpanda-cert1-ca

---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: ECDSA
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 960 bytes and written 363 bytes
Verification: OK
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 256 bit
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
```

Failing test of rest/pandaproxy:
```sh
> k exec -it -n helm-test redpanda-0 -- openssl s_client -connect redpanda-0.redpanda.helm-test.svc.cluster.local.:8082 -CAfile /etc/tls/certs/cert1/ca.crt -tlsextdebug
Defaulted container "redpanda" out of: redpanda, redpanda-configurator (init)
CONNECTED(00000003)
139799559640384:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../ssl/record/ssl3_record.c:331:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 5 bytes and written 283 bytes
Verification: OK
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 0 (ok)
---
command terminated with exit code 1
```

So TLS doesn't seem to be enabled on pandaproxy endpoint, or maybe it is but the certificate isn't actually available. I don't think it's the latter because I would expect kubernetes to complain while trying to start Redpanda if the expect certificates or issuers weren't available.

Another reason is because I'm using the same TLS certificate for all endpoints. This is a big change from what's currently in the main branch. But based I don't see this as being an issue because the endpoints all have the same domain and other details. Only the port is different, and there is no reference to this port in the certificate.

I will try enabling TLS on only rest/pandaproxy next.

**Comment by vuldin** [comment]
This was mainly an issue with sending Redpanda conflicting config through both the redpanda.yaml (via ConfigMap) and the startup flags (via StatefulSet).

**Comment by vuldin** [comment]
Closed in v2 branch with #85
